dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1514,3 +1515,38 @@ CDefFolderMenu_Create2(
 
     return S_OK;
 }
+
+/*
+ * for internal use
+ */
+HRESULT WINAPI
+Shell_DefaultContextMenuCallBack(IShellFolder *psf, IDataObject *pdtobj)
+{
+    PIDLIST_ABSOLUTE pidlFolder;
+    PUITEMID_CHILD *apidl;
+    UINT cidl;
+    HRESULT hr = SH_GetApidlFromDataObject(pdtobj, &pidlFolder, &apidl, &cidl);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    if (cidl > 1)
+        ERR("SHMultiFileProperties is not yet implemented\n");
```

Comment:
What happens if we return an error here?

Reply:
Property sheet doesn't open if selection was multiple.
Memory leaks.

Reply:
OK. It will return E_FAIL.